tests/test_mechaclient.py

```diff
@@ -0,0 +1,168 @@
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE
+"""
+import pytest
+
+from src.mechaclient import MechaClient
+from src.packages.fact_manager import FactManager
+from src.packages.board.rat_board import RatBoard
+from src.packages.cache.rat_cache import RatCache
+
+from src.packages.commands import command
+from src.packages.context.context import Context
+
+pytestmark = pytest.mark.mechaclient
+
+
+def test_verify_version(bot_fx):
+    """
+    Asserts the version is set, and returning string.
+    """
+    assert bot_fx.__version__
+    assert isinstance(bot_fx.__version__, str)
+
+
+def test_verify_api_handler_private(bot_fx):
+    """
+    Asserts the private reference for the API handlers is none, after instantiation.
+    """
+    assert bot_fx._api_handler is None
+
+
+def test_verify_fact_manager_private(bot_fx):
+    """
+    Asserts the private reference for the Fact Manager is none, after instantiation.
+    """
+    assert bot_fx._fact_manager is None
+
+
+def test_verify_rat_cache_private(bot_fx):
+    """
+    Asserts the private reference for the Rat Cache is none, after instantiation.
+    """
+    assert bot_fx._rat_cache is None
+
+
+def test_verify_rat_board_private(bot_fx):
+    """
+    Asserts the private reference for the Rat Board is none, after instantiation.
+    """
+    assert bot_fx._rat_board is None
+
+
+def test_verify_api_handler(bot_fx):
+    """
+    Asserts the api_handler property returns an object, and cannot be set (no setter)
+    """
+    # Helper Class
+    class FakeAPIHandler:
+        ...
+
+    assert bot_fx.api_handler is None
+
+    with pytest.raises(AttributeError):
+        bot_fx.api_handler = FakeAPIHandler()
+
+
+@pytest.mark.parametrize("fact_manager_bad_types", ["Something that isn't a Fact Manager",
+                                                    (1, 3), RatBoard()])
+def test_verify_fact_manager(bot_fx, fact_manager_bad_types):
+    """
+    Asserts the Fact Manager can be set, requires a FactManager object, and returns properly.
+    """
+    # Helper Class
+    class FakeFactManager(FactManager):
+        # Overriding the parent class init to prevent it from creating a DB connection.
+        def __init__(self):
+            ...
+
+    bot_fx.fact_manager = FakeFactManager()
+    assert isinstance(bot_fx._fact_manager, FakeFactManager)
+    assert bot_fx.fact_manager is bot_fx._fact_manager
+
+    del bot_fx.fact_manager
+    assert bot_fx._fact_manager is None
+
+    with pytest.raises(TypeError):
+        bot_fx.fact_manager = fact_manager_bad_types
+
+
+@pytest.mark.parametrize("rat_board_bad_types", ["Something that isn't a Rat Board", (1, 3), 3.14])
+def test_verify_rat_board(bot_fx, rat_board_bad_types):
+    """
+    Asserts the Rat Board can be set, requires a Rat Board object, and returns properly.
+    """
+    bot_fx.board = RatBoard()
+
+    assert isinstance(bot_fx._rat_board, RatBoard)
+    assert bot_fx.board is bot_fx._rat_board
```

Comment:
[Discussion] Should these tests be separated? Given the parametrize, it's testing both a good path and a bad path with each invocation.

Reply:
They could be separated, but what would be gained from splitting it?  The tests do pass with known good or bad data.

Reply:
It's more of a convention thing IMO, making it so one test tests one thing. No real tangible benefit other than organization. (Which is still nice to have.)

Reply:
Arguably, tests should only test one scenario

Either test the good path, or the bad path, but not both within a test definition.